src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/services/OffenceCodeLookupService.kt

```diff
@@ -0,0 +1,179 @@
+package uk.gov.justice.digital.hmpps.hmppsmanageadjudicationsapi.services
+
+import org.springframework.stereotype.Service
+
+@Service
+class OffenceCodeLookupService {
+  var offenceCodeDetailsByCode: Map<Int, OffenceCodeDetails> = OffenceCodeLookupBuilder().buildlookup()
+
+  fun getCommittedOnOwnNomisOffenceCodes(offenceCode: Int): List<String> {
+    return offenceCodeDetailsByCode[offenceCode]?.nomisCodes ?: OffenceCodeDefaults.DEFAULT_NOMIS_IDS
+  }
+
+  fun getNotCommittedOnOwnNomisOffenceCode(offenceCode: Int): String {
+    return offenceCodeDetailsByCode[offenceCode]?.notCommittedOnOwnNomisCode ?: OffenceCodeDefaults.DEFAULT_NOMIS_ID
+  }
+
+  fun getParagraphNumber(offenceCode: Int): String {
+    return offenceCodeDetailsByCode[offenceCode]?.paragraph?.paragraphNumber ?: OffenceCodeDefaults.DEFAULT_PARAGRAPH_DATA
+  }
+
+  fun getParagraphDescription(offenceCode: Int): String {
+    return offenceCodeDetailsByCode[offenceCode]?.paragraph?.paragraphDescription ?: OffenceCodeDefaults.DEFAULT_PARAGRAPH_DATA
+  }
+}
+
+class OffenceCodeLookupBuilder {
+  private var allCodeBuilders: MutableList<OffenceCodeLookupCodeBuilder> = mutableListOf()
+
+  // This information is taken from the spreadsheet of offences related to the ticket this work was done under
+  fun buildlookup(): Map<Int, OffenceCodeDetails> {
```

Comment:
I started with a basic map, but it wasn't succinct enough so i did this..

Reply:
I do wonder if at some point the questions should be pushed into the back end - we could keep the data structure in the front end as it is - just read the data into it from the back end. Anyway that a story for another day